extensions/clusters/clusters.go

```diff
@@ -868,32 +856,31 @@ func HardenK3SClusterConfig(clusterName, namespace string, clustersConfig *Clust
 	return v1Cluster
 }
 
-// HardenRKE2ClusterConfig is a constructor for a apisV1.Cluster object, to be used by the rancher.Client.Provisioning client.
+// HardenRKE1ClusterConfig is a function that modifies the cluster configuration to be hardened according to the CIS benchmark.
+func HardenRKE1ClusterConfig(client *rancher.Client, clusterName string, clustersConfig *ClusterConfig) *management.Cluster {
+	cluster := NewRKE1ClusterConfig(clusterName, client, clustersConfig)
+
+	cluster.DefaultPodSecurityAdmissionConfigurationTemplateName = rancherRestricted
+	cluster.RancherKubernetesEngineConfig.Services.Etcd.GID = 52034
+	cluster.RancherKubernetesEngineConfig.Services.Etcd.UID = 52034
+
+	return cluster
+}
+
+// HardenRKE2ClusterConfig is a function that modifies the cluster configuration to be hardened according to the CIS benchmark.
 func HardenRKE2ClusterConfig(clusterName, namespace string, clustersConfig *ClusterConfig, machinePools []apisV1.RKEMachinePool, cloudCredentialSecretName string) *apisV1.Cluster {
 	v1Cluster := NewK3SRKE2ClusterConfig(clusterName, namespace, clustersConfig, machinePools, cloudCredentialSecretName)
+	v1Cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName = rancherRestricted
 
-	if clustersConfig.KubernetesVersion <= string(provisioninginput.PSPKubeVersionLimit) {
-		v1Cluster.Spec.RKEConfig.MachineSelectorConfig = []rkev1.RKESystemConfig{
-			{
-				Config: rkev1.GenericMap{
-					Data: map[string]interface{}{
-						"profile":             "cis-1.6",
-						protectKernelDefaults: true,
-					},
-				},
-			},
-		}
-	} else {
-		v1Cluster.Spec.RKEConfig.MachineSelectorConfig = []rkev1.RKESystemConfig{
-			{
-				Config: rkev1.GenericMap{
-					Data: map[string]interface{}{
-						"profile":                 "cis-1.23",
-						"protect-kernel-defaults": true,
-					},
+	v1Cluster.Spec.RKEConfig.MachineSelectorConfig = []rkev1.RKESystemConfig{
+		{
+			Config: rkev1.GenericMap{
+				Data: map[string]interface{}{
+					"profile":             "cis",
```

Comment:
is this going to change frequently? Just noticed that this is `cis`, it used to be `cis-1.6` or `cis-1.23`, and the test name is `cis 1.8`. Should this be read in from the scan name or something instead of hard coding it?

Reply:
`cis` specifically was introduced to address this concern. Moving forward, `cis` is what is documented that users should be using so that with future profiles, this won't change.

Upgrading as an example is a huge plus as to keeping this.